pkg/webhook/resources/setting/validator.go

```diff
@@ -709,6 +723,159 @@ func (v *settingValidator) validateDeleteStorageNetwork(setting *v1beta1.Setting
 	return werror.NewMethodNotAllowed(fmt.Sprintf("Disallow delete setting name %s", settings.StorageNetworkName))
 }
 
+func (v *settingValidator) getPrometheusVolumes() ([]string, error) {
+	prometheusVolumes := make([]string, 0)
+	sets := labels.Set{
+		labelAppNameKey: labelAppNameValuePrometheus,
+	}
+
+	pvcs, err := v.pvcCache.List(util.CattleMonitoringSystemNamespace, sets.AsSelector())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, pvc := range pvcs {
+		if pvc.Spec.VolumeName == "" {
+			continue
+		}
+
+		prometheusVolumes = append(prometheusVolumes, pvc.Spec.VolumeName)
+	}
+
+	return prometheusVolumes, nil
```

Comment:
Those are redundant codes, prefer to create a common method to pass the label and get PVCs result accordingly.